src/idl_gen_general.cpp

```diff
@@ -361,6 +361,29 @@ static std::string DestinationValue(const LanguageParameters &lang,
   }
 }
 
+// Cast statements for mutator method parameters.
+// In Java, parameters representing unsigned numbers need to be cast down to their respective type.
+// For example, a long holding an unsigned int value would be cast down to int before being put onto the buffer.
+// In C#, one cast directly cast an Enum to its underlying type, which is essential before putting it onto the buffer.
+static std::string SourceCast(const LanguageParameters &lang,
```

Comment:
No, just wanted to make sure you were aware of it. If merging the two gets too convoluted, leave as-is.